client/session.go

```diff
@@ -170,8 +192,38 @@ func (s *session) setTopologyMap(topologyMap m3db.TopologyMap) error {
 	s.Lock()
 	prev := s.queues
 	s.queues = queues
-	s.setWriteOpPoolsWithLock(topologyMap)
 	s.topoMap = topologyMap
+	prevReplicas := s.replicas
+	s.replicas = topologyMap.Replicas()
+	s.quorum = topologyMap.QuorumReplicas()
+	if s.fetchBatchOpArrayArrayPool == nil ||
+		s.fetchBatchOpArrayArrayPool.Entries() != len(queues) {
+		s.fetchBatchOpArrayArrayPool = newFetchBatchOpArrayArrayPool(
+			s.opts.GetFetchBatchOpPoolSize(),
+			len(queues),
+			s.opts.GetFetchBatchOpPoolSize()/s.opts.GetFetchBatchSize())
```

Comment:
what's the rationale behind this? Seems the formula is something along the lines of:
```
total number of fetchBatchOps == fetchBatchOpPoolSize == fetchBatchArrayArrayPoolSize * number of hosts (i.e., len(queues)) * capacity
```
which means
```
capacity =  fetchBatchOpPoolSize / (fetchBatchOpArrayArrayPoolSize * len(queues))
```
Is this correct?